channels/serializers.py

```diff
@@ -475,6 +498,13 @@ def update(self, instance, validated_data):
             elif validated_data["subscribed"] is False:
                 api.remove_post_subscription(post_id)
 
+        cover_image = validated_data.get("cover_image")
+        if instance and cover_image and hasattr(instance, "article"):
+            instance.article.cover_image.save(
+                f"article_image_{post_id}.jpg", cover_image, save=False
+            )
+            instance.article.save(update_fields=["cover_image"], update_image=True)
```

Comment:
I'm wondering, should this logic for updating the article cover image maybe live in the `update_post` method on the `api` class? I'm just wondering because with this setup if you update an article post we're going to end up calling `.save` twice, once above in the call to `api.update_post` and again down here. Maybe it would be better to have this logic live in that method instead, so it's both all in one place and so that we can just call `.save` once?

Reply:
✔️

Reply:
Oops, sorry, fixed now.